pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -179,18 +203,29 @@ func (d *PmemCSIDriver) reconcileDeploymentChanges(r *ReconcileDeployment, exist
 		}
 	}
 
-	if updateController {
-		klog.Infof("Updating controller driver for deployment %q", d.Name)
-		if e := r.Update(d.getControllerStatefulSet()); e != nil {
-			requeue = true
-			err = e
+	// Force update all deployment objects
+	if updateAll {
+		klog.Infof("Updating all objects for deployment %q", d.Name)
```

Comment:
Please add a test for this case. A thorough test ensures that all objects are slightly different than they should be before the operator starts and afterwards pass the `validateCSIDriver` function.

That function itself is not as thorough as it should be (only checks some fields, but not everything); you can ignore that for now, I am working on it.

Reply:
Added a test that:
  * stats a deployment with defaults
  * removes operator pod
  * updates all deployment fields
  * restarts pod
  * wait till the deployment timestamp gets updated and ensures that all new values are applied to its sub-objects.